sdk/core/azure-core-management/src/main/java/com/azure/core/management/Resource.java

```diff
@@ -62,4 +67,46 @@ public Resource withTags(Map<String, String> tags) {
         this.tags = tags;
         return this;
     }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        return jsonWriter.writeStartObject()
+            .writeStringField("location", location)
+            .writeMapField("tags", tags, JsonWriter::writeString)
+            .writeEndObject();
+    }
+
+    /**
+     * Reads a JSON stream into a {@link Resource}.
+     *
+     * @param jsonReader The {@link JsonReader} being read.
+     * @return The {@link Resource} that the JSON stream represented, may return null.
+     * @throws IOException If a {@link Resource} fails to be read from the {@code jsonReader}.
+     */
+    public static Resource fromJson(JsonReader jsonReader) throws IOException {
+        return jsonReader.readObject(reader -> {
+            Resource resource = new Resource();
+
+            while (reader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = reader.getFieldName();
+                reader.nextToken();
+
+                if ("id".equals(fieldName)) {
+                    ProxyResourceAccessHelper.setId(resource, reader.getString());
+                } else if ("name".equals(fieldName)) {
+                    ProxyResourceAccessHelper.setName(resource, reader.getString());
+                } else if ("type".equals(fieldName)) {
+                    ProxyResourceAccessHelper.setType(resource, reader.getString());
```

Comment:
We can do it here. But seems won't able to do such in a subclass of `Resource`.

Reply:
Yeah, I was thinking about adding that constructor. Though would we want to share that through all implementations, requiring these three values to be passed through the constructor.